kuka_rsi_hw_interface/krl/README.md

```diff
@@ -1,4 +1,93 @@
-# KRL CODE
+# Configuring RSI on the controller
+
+This guide highlights the steps needed in order to successfully configure the **RSI interface** on the controller to work with the **kuka_rsi_hardware_interface** on your PC with ROS.
+
+## 1. Controller network configuration
+
+Windows runs behind the SmartHMI on the teach pad. Make sure that the **Windows interface** of the controller and the **PC with ROS** is connected to the same subnet.
+
+1. Log in as **Administrator** on the teach pad and navigate to **Network configuration** (Main menu -> Start-up -> Network configuration -> Activate advanced configuration).
+2. There should already be an interface checked out as the **Windows interface**.
+   * **IP**: e.g. 192.168.250.20
+   * **Subnet mask**: 255.255.255.0
+   * **Default gateway**: 192.168.250.20
+   * **Windows interface checkbox** should be checked.
+3. Minimize the SmartHMI (Main menu -> Start-up -> Service -> Minimize HMI).
+4. Run the application called **RSI-Network** from the Windows desktop environment on the teach pad.
+5. Check that the **Network - Kuka User Interface** show the Windows interface with the specified IP address.
+6. Add a new IP address on another subnet (e.g. 192.168.1.20) for the **RSI interface**.
+   * This should be possible within **RSI-Network**, if not, then do it within the SmartHMI at **Network configuration**.
+7. Reboot the controller (Main menu -> Shutdown -> Check **Force cold start** and **Reload files** -> Reboot control PC).
+8. After reboot, minimize the SmartHMI (Main menu -> Start-up -> Service -> Minimize HMI).
+9. Run **cmd.exe** and ping the PC you want to communicate with on the same subnet (e.g. 192.168.250.xx).
+
+If your **PC** has an IP address on the same subnet as the **Windows interface** on the controller, the controller should receive answers from the PC:
+* If this is the case, add another IP address to the current PC connection (e.g. 192.168.1.xx) on the same subnet as the **RSI** interface.
+
+## 2. KRL Files
+
+The files included in this folder specifies the data transferred via RSI. Some of the files needs to be modified to work for your specific configuration.
+
+##### ros_rsi_ethernet.xml
+1. Edit the `IP_NUMBER` tag so that it corresponds to the IP address (192.168.1.xx) previously added for your PC.
+2. Keep the `PORT` tag as it is (49152) or change it if you want.

```

Comment:
It's mentioned later on, but it might be good to state here already that if the port is changed, the same value should be used for the `rsi/port` parameter.